pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -44,30 +72,73 @@ def sipopt(instance, paramSubList, perturbList,
 
     keepfiles: bool, optional
         preserve solver interface files
+            
+    optarg : dictionary, optional
+        ipopt solver options dictionary object for method = 'kaug' (default=None)
     
     Returns
     -------
     model: ConcreteModel
-        The model modified for use with sipopt.  The returned model has
-        three :class:`Suffix` members defined:
+        if method is 'sipopt',
+        The model is modified for use with sipopt.  The returned model has
+            three :class:`Suffix` members defined:
 
         - ``model.sol_state_1``: the approximated results at the
           perturbation point
         - ``model.sol_state_1_z_L``: the updated lower bound
         - ``model.sol_state_1_z_U``: the updated upper bound
-
+        
+        if method is 'kaug',
+        The model is modified for use with kaug.  
+        The model contains the approximated results at the perturbation point
+        
     Raises
     ------
+    ValueError
+        method argument should be either 'sipopt' or 'kaug'
     ValueError
         perturbList argument is expecting a List of Params
     ValueError
         length(paramSubList) must equal length(perturbList)
     ValueError
         paramSubList will not map to perturbList
-
+    ImportError
+        ipopt binary must be available
+    ImportError
+        k_aug binary must be available
+    ImportError
+        dotsens binary must be available
+    Exception
+        kaug does not support inequality constraints
     """
-
-    #Verify User Inputs    
+    # Verify User Inputs    
+    if method == 'sipopt':
+        opt = SolverFactory('ipopt_sens', solver_io='nl')
```

Comment:
Instead of `opt`, why not call this variable `sipopt` or `ipopt_sens` to match what you are doing for `kaug` below?

Reply:
I changed them.